tests/generics/tpointerprocs.nim

```diff
@@ -0,0 +1,28 @@
+discard """
+cmd: "nim check $options --hints:off $file"
+action: "reject"
+nimout:'''
+tpointerprocs.nim(15, 11) Error: 'foo' doesn't have a concrete type, due to unspecified generic parameters.
+tpointerprocs.nim(27, 11) Error: cannot instantiate: 'foo[int]'; got 1 typeof(s) but expected 2
+tpointerprocs.nim(27, 14) Error: expression 'foo[int][int]' has no type (or is ambiguous)
```

Comment:
Why `foo[int][int]`?

Reply:
It's nim check giving out all the errors, so `foo[int]` got turned into `foo[int][int]`, should I change the error max to the number of expected errors?

Reply:
Well yes, I know but why got it turned into this?

Reply:
So it looks like this is due to https://github.com/nim-lang/Nim/blob/30d28bcefcad0da8900cfa231be9d77bb98c5097/compiler/semexprs.nim#L1513-L1516
What's done in the if statements adds the `[int]` on it again.